file_management/include/file_management/file_manager_options.h

```diff
@@ -35,12 +35,14 @@ struct FileManagerStrategyOptions {
     std::string _file_prefix,
     std::string _file_extension,
     size_t _maximum_file_size,
-    size_t _storage_limit)
+    size_t _storage_limit,
+    bool _discard_old_logs = false)
```

Comment:
```suggestion
    bool _discard_old_data = false)
```

If we are making this a generic option then the time limit for the data should be generic as well. Otherwise it's hardcoded to the CWL API limit.

I think we should re-evaluate refactoring this logic such that it happens only in the Log or Metrics implementation. Maybe the tradeoff of file I/O in the readBatch function is fine in this case. 

@mm318 what do you think?

Reply:
I think the discard should be a configurable option for both. We can do 14 days for CloudWatch Logs by default, and infinite (or something else) for CloudWatch Metrics by default.